File: app/src/main/java/com/inqint/yarnrequirements/Projects/LongLengthUnits.java
package com.inqint.yarnrequirements.Projects;

/* Enumeration for longer lengths: yards or meters */
public enum LongLengthUnits { yards, meters;
    private static LongLengthUnits[] values = null;
    public static LongLengthUnits fromInt(int i) {
        if (LongLengthUnits.values == null) {
            LongLengthUnits.values = LongLengthUnits.values();
        }
        return LongLengthUnits.values[i];
    }
}

File: app/src/main/java/com/inqint/yarnrequirements/Projects/LongLengthUnits.kt
package com.inqint.yarnrequirements.Projects
import java.io.Serializable

/* Enumeration for longer lengths: yards or meters */
enum class LongLengthUnits: Serializable {
    yards, meters;


    companion object {
        private var values: Array<LongLengthUnits>? = null
        fun fromInt(i: Int): LongLengthUnits {
            if (LongLengthUnits.values == null) {
                LongLengthUnits.values = LongLengthUnits.values()
            }
            return LongLengthUnits.values!![i]
        }
    }
}
